### DISCUSSION TITLE
API Power Automate integration

### DISCUSSION THREAD

**Discussion by schwibach** [SchoolUtils/WebUntis#59] [Ideas]
Hello,

I'm curious if it would be (easily) possible to query the webunits database with power automate using the API.
It would be quite interesting to ETL Class lists etc. into M365.

**Comment by dunklesToast** [comment]
Hey,

since you [can do HTTP Requests inside Power Automate](https://techcommunity.microsoft.com/t5/microsoft-365-pnp-blog/how-to-get-started-with-http-requests-in-power-automate/ba-p/2121908) it should be possible. However I've never worked with Power Automate so I can't say if it's easily doable.

**Reply by schwibach** [reply]
I have done HTTP requests in Power Automate in the past. I can handle that end.
However, I don't know how to structure it, as in what the request would have to look like in order to get something usable.

Do you have any pointers where to find infos on what headers, queries, body and authentication should look like?
<img width="415" alt="power automate http" src="https://user-images.githubusercontent.com/2668466/149834856-69513661-0eef-4703-bb4b-68d58f730629.png">

**Reply by luwol03** [reply]
You have two (three) options:
1. reverse enginiering the SchoolUntis/WebUntis project (or any other API wrapper)
2. Write an email to the untis helpdesk center that you would like to get the API documentation. I looked for a link to that discussion where that method was mentioned, but it seems like they closed their forum.
3. *google*

I did the second one. And it was very easy. It just took some time.

Hope that helps.
luwol03

**Reply by schwibach** [reply]
Thanks!
I tried 3 last night, was unsuccessful and tried 2 today.
We'll see what comes of it.

Have you managed to get all the data you wanted? Have you tried it with M365?

**Reply by luwol03** [reply]
I tried 3 last night and I was successful, but I think Im not allowed to paste a link to that here. But I wish you good luck for the second option.

I got everything, what I wanted, but I havent ever used M365.

**Reply by forsterarnaud** [reply]
Hello all, I'm trying to connect to Untis using Power Automate but I get some trouble, too building the http requests.
Did you find a way to make it ? 
Thanks to all for your help
